pkg/controller/account/account_controller.go

```diff
@@ -203,6 +219,12 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, nil
 	}
 
+	awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, setupClientInput)
```

Comment:
I think a comment above here could be nice so we don't "forget" why we only build the client so late. just an idea.